### ISSUE TITLE
1006 for expression bodies

### ISSUE THREAD

**Issue by jnyrup** [u2uconsult/codeanalyzers#26]
`U2U1006` is triggered on the following:
```c#
private static Func<Task<TIn>, Task> AsyncWrapper<TIn>(Func<TIn, Task> func) => 
    async (input) => await func(await input);
```

Removing `async` and the outer `await` results in invalid code.
```c#
private static Func<Task<TIn>, Task> AsyncWrapper<TIn>(Func<TIn, Task> func) =>
    (input) => func(await input);
```

When expanding the inner method body, `U2U1006` is no longer triggered as expected.
```c#
private static Func<Task<TIn>, Task> AsyncWrapper<TIn>(Func<TIn, Task> func) =>
    async (input) =>
    {
        await func(await input);
    };
```

**Comment by KrisVandermotten** [comment]
Thanks for reporting this. This bug is fixed in version 2.18.3, available now.
